File: backend/src/main/java/com/hilbert/features/article/service/ArticleService.java
package com.hilbert.features.article.service;

import com.hilbert.features.article.dto.ArticleFullDto;
import com.hilbert.features.article.dto.ArticleSearchDto;
import com.hilbert.features.article.dto.CreateArticleDto;
import com.hilbert.features.article.dto.UpdateArticleDto;
import com.hilbert.shared.search.models.ArticleSearchParams;
import com.hilbert.shared.search.models.PaginatedResults;

public interface ArticleService {

    ArticleFullDto getArticleById(Long id);
    PaginatedResults<ArticleSearchDto> searchArticles(ArticleSearchParams searchParams);
    ArticleFullDto createArticle(CreateArticleDto articleDto);
    ArticleFullDto updateArticle(UpdateArticleDto articleDto);
    void deleteArticle(Long id);
}

File: mobile-frontend/src/app/features/article/services/ArticleService.ts
import axiosInstance from "../../../core/main/config/AxiosInstance";
import { ArticleSearchParams, PaginatedResults } from "../../../shared/search/models/Search";
import { SearchUrlService } from "../../../shared/search/services/SearchUrlService";
import { ArticleFullDto, ArticleSearchDto, CreateArticleDto, UpdateArticleDto } from "../models/Article";

const RESOURCE_PATH = "/articles";

export const ArticleService = {
    
    searchArticles: async (searchParams: ArticleSearchParams): Promise<PaginatedResults<ArticleSearchDto>> => {
        const searchUrlSuffix = RESOURCE_PATH + "/search" + 
            SearchUrlService.constructSearchUrl(searchParams); 

        const response = await axiosInstance.get<PaginatedResults<ArticleSearchDto>>(searchUrlSuffix);
        return response.data;
    },

    getArticle: async (articleId: number): Promise<ArticleFullDto> => {
        const response = await axiosInstance.get<ArticleFullDto>(`${RESOURCE_PATH}/${articleId}`);
        return response.data;
    },

    createArticle: async (articleDto: CreateArticleDto): Promise<ArticleFullDto> => {
        const response = await axiosInstance.post<ArticleFullDto>(RESOURCE_PATH, articleDto);
        return response.data;
    },

    updateArticle: async (articleDto: UpdateArticleDto): Promise<ArticleFullDto> => {
        const response = await axiosInstance.put<ArticleFullDto>(RESOURCE_PATH, articleDto);
        return response.data;
    },

    deleteArticle: async (articleId: number): Promise<void> => {
        axiosInstance.delete(`${RESOURCE_PATH}/${articleId}`);
    }
}